chapters/traversal.md

```diff
@@ -1,83 +1,89 @@
+print(len(root.chunks))                # outputs 0
+
+# Initialise two Documents and add them as chunks to root.
 with Document() as chunk1:
     chunk1.text = 'What is love?'
     root.chunks.add(chunk1)
 with Document() as chunk2:
     chunk1.text = 'Oh baby do not hurt me.'
     root.chunks.add(chunk2)
-# Now the document has 2 chunks
-print(len(root.chunks))
->>> 2
+
+# Now the Document has 2 chunks
+print(len(root.chunks))                # outputs 2
 ```
 
-What happened by adding `chunk` to `root`?
+### Internals: Adding a chunk to a root node
+
+Now let us have a look at what happens when you added a chunk to a root node. The code example below shows the granularity level of a specific node using the attribute `granularity` -- as a root node (level 0), and as a chunk (level 1).
 
 ```python
-print(root.granularity)
->>> 0
-print(root.chunks[0].granularity)
->>> 1
-root.id == root.chunks[0].parent_id
->>> True
+print(root.granularity)                # outputs 0
+print(root.chunks[0].granularity)      # outputs 1
 ```
-
-This can be seen in the image below:
+This is illustrated in the image below:
 
 ![granularity](./images/granularity.png)
 
-The code sample and graph above demonstrates the basic idea of a `chunk` in a Document.
-In the beginning, we initialized a Document with `granularity=0` (by default).
-We then initialized two chunks and add them to the `root` document.
-Two things happened when adding the `chunk` to `root`:
+Furthermore, with the help of the attribute `parent_id` the relation between the root node and the chunk can be validated.
+
+```python
+print(root.id == root.chunks[0].parent_id)  # outputs True
+```
+
+The code sample and graph above demonstrates the basic idea of a `chunk` in a Document. At the beginning, by default a Document is initialised with `granularity = 0`. Next, two chunks have been initialised with `granularity = 0`, and have been added to the root node. Adding a chunk to a node, two things happen:
 
-1. The granularity of the chunk has been increased by 1 (default 0).
-2. The `chunk` has been referenced to its parent: `root`.
+1. The granularity of the chunk will be increased by 1.
+2. The chunk will be referenced to its parent which is the root node.
 
-This allows Jina (and you) to query chunks and reference back to its root document at any `granularity` level.
+This procedure allows Jina (and you) to query chunks and reference them back to its root Document at any `granularity` level.
 
 ## Matches
 
-In a neural search system (and traditional retrieval system), matches are the expected documents returned from the system given the user query.
-In Jina, `matches` could happen at the `root` level or any `chunk` level.
+In a neural search system (and traditional retrieval system), *matches* are the expected Documents returned from the search index given the user query. In Jina, matches could happen at any level of the representation tree -- the root level or any chunk level.
```

Comment:
`change index given the user query` to `change index for the user query`

Reply:
ok, I have changed that, too